pkg/util/gitutil/git.go

```diff
@@ -55,48 +70,85 @@ func GetGitRepository(dir string) (*git.Repository, error) {
 
 // GetGitHubProjectForOrigin returns the GitHub login, and GitHub repo name if the "origin" remote is
 // a GitHub URL.
-func GetGitHubProjectForOrigin(dir string) (string, string, error) {
+func GetGitHubProjectForOrigin(dir string) (string, string, string, error) {
 	repo, err := GetGitRepository(dir)
 	if repo == nil {
-		return "", "", fmt.Errorf("no git repository found from %v", dir)
+		return "", "", "", fmt.Errorf("no git repository found from %v", dir)
 	}
 	if err != nil {
-		return "", "", err
+		return "", "", "", err
 	}
-	return GetGitHubProjectForOriginByRepo(repo)
+	remoteURL, err := GetGitRemoteURL(repo, "origin")
+
+	if err != nil {
+		return "", "", "", err
+	}
+	return TryGetVCSInfo(remoteURL)
 }
 
-// GetGitHubProjectForOriginByRepo returns the GitHub login, and GitHub repo name if the "origin" remote is
-// a GitHub URL.
-func GetGitHubProjectForOriginByRepo(repo *git.Repository) (string, string, error) {
-	remote, err := repo.Remote("origin")
+// GetGitRemoteURL returns the remote URL for the given remoteName in the repo.
+func GetGitRemoteURL(repo *git.Repository, remoteName string) (string, error) {
+	remote, err := repo.Remote(remoteName)
 	if err != nil {
-		return "", "", errors.Wrap(err, "could not read origin information")
+		return "", errors.Wrap(err, "could not read origin information")
 	}
 
 	remoteURL := ""
 	if len(remote.Config().URLs) > 0 {
 		remoteURL = remote.Config().URLs[0]
 	}
+
+	return remoteURL, nil
+}
+
+// IsGitOriginURLGitHub returns true if the provided remoteURL is detected as GitHub.
+func IsGitOriginURLGitHub(remoteURL string) bool {
+	return strings.Contains(remoteURL, GitHubHostName)
+}
+
+// TryGetVCSInfo attempts to detect whether the provided remoteURL
+// is an SSH or an HTTPS remote URL. It then extracts the repo, owner name,
+// and the type (kind) of VCS from it.
+func TryGetVCSInfo(remoteURL string) (string, string, string, error) {
 	project := ""
+	vcsKind := ""
 
-	const GitHubSSHPrefix = "git@github.com:"
-	const GitHubHTTPSPrefix = "https://github.com/"
-	const GitHubRepositorySuffix = ".git"
+	if cloudSourceControlSSHRegex.MatchString(remoteURL) {
+		// Get all matching groups.
+		matches := cloudSourceControlSSHRegex.FindAllStringSubmatch(remoteURL, -1)[0]
+		// Get the named groups in our regex.
+		groupNames := cloudSourceControlSSHRegex.SubexpNames()
 
-	if strings.HasPrefix(remoteURL, GitHubSSHPrefix) {
-		project = trimGitRemoteURL(remoteURL, GitHubSSHPrefix, GitHubRepositorySuffix)
-	} else if strings.HasPrefix(remoteURL, GitHubHTTPSPrefix) {
-		project = trimGitRemoteURL(remoteURL, GitHubHTTPSPrefix, GitHubRepositorySuffix)
+		groups := map[string]string{}
+		for i, value := range matches {
+			groups[groupNames[i]] = value
+		}
+		vcsKind = groups["host_name"]
+		project = groups["owner_and_repo"]
+		project = strings.TrimSuffix(project, defaultGitCloudRepositorySuffix)
+	} else {
```

Comment:
A comment to explain the difference between the `cloudSourceControlSSHRegex` and this `else block` would be helpful.

To make the code clearer, I'd even go so far as to do something like:

```
// If the remote is using SSH, we strip out the relevant data via regex.
if strings.HasPrefix(remoteURL, "git@") {
     ...
}

// For HTTP(S) based remotes, we can just rely on the URL itself.
if strings.hasPrefix(remoteURL, "http") {
    ...
}

Reply:
I have updated the conditionals a bit and added some clarifying comments as well.